### PR TITLE
On GitHub Actions, run only the e2e tests that are reliable there

### PR DESCRIPTION
**Pull request by philrz**
## tl;dr

For the "Advance Zed" workflow, this PR proposes trimming the set of e2e tests to only those that have a history of running reliably in GitHub Actions.

To test it, I created a separate workflow that just runs the e2e tests. Rather than scrapping it, I'd propose keeping it around since exercises like this have come up a couple times in the past and it seems a little silly to keep recreating it every time.

## Details

The `title-bar-buttons.spec.ts` test that was just added as part of #2953 seems to run reliably locally on macOS, but in [first attempts to run in Actions](https://github.com/brimdata/zui/actions/runs/7425676367) it failed 5 times out of 5 (`Test timeout of 30000ms exceeded. / Error: locator.click: Application exited`). While we could invest cycles into seeing why this particular test doesn't work in Actions, it can also be seen as another in a series of "Playwright doesn't run reliably enough on CI" data points.

This poses a challenge because the e2e tests have traditionally been the best at catching when the app breaks due to unexpected changes in Zed. In the past when before we ran e2e tests as a prerequisite to advancing the Zed pointer, sometimes multiple breakages would stack atop each other and this would become very difficult to untangle. Therefore in theory we'd like to have them running frequently. However, their perceived flakiness has had the effect of desensitizing us from failures. We've also often wasted time smashing **Re-run** just for the satisfaction of seeing the green status and advancing the pointer. In effect, the "automation" became less automated.

While we've toyed with the idea of turning off the e2e tests in Actions and running them exclusively by hand, @nwt proposed that we see if we can just eliminate the unreliable ones. In preparation for this PR, I did an audit of the last 3 months of "Advance Zed" run outputs (that's as far back as it goes) and have concluded that we could eliminate the bad runs if we skip `pool-loads.spec.ts` and `pool-groups.spec.ts` in addition to the just-added `title-bar-buttons.spec.ts`.

## Audit & Observations

Here's a summary of which failures-due-to-apparent-unreliability we had in which runs.

	• https://github.com/brimdata/zui/actions/runs/7427850158
		○ pool-loads.spec.ts:42:7 › Pool Loads › bad data displays an error message
		○ title-bar-buttons.spec.ts:9:3 › title bar buttons › toggle left sidebar
		○ title-bar-buttons.spec.ts:16:3 › title bar buttons › toggle right sidebar
	• https://github.com/brimdata/zui/actions/runs/7425676367
		○ pool-loads.spec.ts:42:7 › Pool Loads › bad data displays an error message
		○ title-bar-buttons.spec.ts:9:3 › title bar buttons › toggle left sidebar
		○ title-bar-buttons.spec.ts:16:3 › title bar buttons › toggle right sidebar
	• https://github.com/brimdata/zui/actions/runs/7424306865
		○ pool-loads.spec.ts:42:7 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/7360182387
		○ pool-groups.spec.ts:63:7 › Pool Groups › delete group
		○ pool-loads.spec.ts:42:7 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/7050085559
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/7038166640
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6961214026
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6910207870
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6897010755
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6868975835
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6830941052
		○ pool-groups.spec.ts:33:3 › Pool Groups › edge cases
	• https://github.com/brimdata/zui/actions/runs/6777567934
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6763656722
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6755689465
		○ pool-groups.spec.ts:33:3 › Pool Groups › edge cases
	• https://github.com/brimdata/zui/actions/runs/6755688207
		○ pool-groups.spec.ts:33:3 › Pool Groups › edge cases
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6746521793
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6671607413
		○ pool-loads.spec.ts:42:3 › Pool Loads › bad data displays an error message
	• https://github.com/brimdata/zui/actions/runs/6565561412
		○ pool-groups.spec.ts:44:3 › Pool Groups › rename group

### `pool-groups.spec.ts`

In the past I've eyeballed failures such as the one below:

```
1) pool-groups.spec.ts:44:3 › Pool Groups › rename group =========================================

    Error: expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 1

      Array [
    +   "backups / yesterday",
        "bkups / today",
    -   "bkups / yesterday",
      ]

      55 |
      56 |     const pools = await app.zealot.getPools()
    > 57 |     expect(pools.map((p) => p.name).sort()).toEqual([
         |                                             ^
      58 |       "bkups / today",
      59 |       "bkups / yesterday",
      60 |     ])

        at /home/runner/work/zui/zui/packages/e2e-tests/tests/pool-groups.spec.ts:57:45
        at runMicrotasks (<anonymous>)
```

To which I observed (and @jameskerr agreed):

> So it would imply that either only some of the pool renames succeeded or the test is checking the contents of the updated pools list too fast. Given problems we've had in the past, I'm inclined to suspect the latter.

Therefore perhaps at some point we could work to make that one reliable and bring it back into the mix.

### `pool-loads.spec.ts`

The pattern of failures here remains mysterious. One observation I'll make is that it _is_ capable of running reliably, e.g., it ran reliably on Actions through pretty much the whole month of December. But then sometimes we'll see it fail multiple times close together. One suspicion we've always had about Actions is that the shared nature of the infrastructure just makes it prone to slowness or limited resources resulting in processes getting shot, so maybe its bad runs just correlate with bad times in Actions overall.

image

### `title-bar-buttons.spec.ts`

100% a mystery right now.

### Others (not shown above)

There were several legit failures observed in the Export VNG test because that area of Zed has been undergoing intentional changes in the format that affect the file size. Given the frequency of failures we've toyed with the idea of temporarily disabling that test until VNG stabilizes, but really, to me it illustrates the very benefit we want from these tests: I observe the failure, confirm the change was unsurprising, adapt the test, and move foward.

There also were a couple failures surely due to things like temporary network outages, e.g., timing out when downloading some dependency or uploading an artifact at the end of a run. That just life in Actions, baby!

### CODE REVIEW

**Comment by jameskerr** [comment]
Hi @philrz, I'm going to add a commit that re-arranges where we keep the list of flaky tests. I'm going to put it in the package.json of the zui-player and use the --grep-invert option that we can pass to the playwright test command.

Here's an example of the regex working to isolate those test files:

```
❯ yarn test --grep '(pool-loads|pool-groups|title-bar-buttons).spec' --list
Listing tests:
  pool-groups.spec.ts:22:7 › Pool Groups › groups pools by slash
  pool-groups.spec.ts:33:7 › Pool Groups › edge cases
  pool-groups.spec.ts:44:7 › Pool Groups › rename group
  pool-groups.spec.ts:63:7 › Pool Groups › delete group
  pool-loads.spec.ts:17:7 › Pool Loads › load data into a pool
  pool-loads.spec.ts:27:7 › Pool Loads › load more data into the pool
  pool-loads.spec.ts:42:7 › Pool Loads › bad data displays an error message
  title-bar-buttons.spec.ts:4:3 › title bar buttons › click new tab
  title-bar-buttons.spec.ts:9:3 › title bar buttons › toggle left sidebar
  title-bar-buttons.spec.ts:16:3 › title bar buttons › toggle right sidebar
Total: 10 tests in 3 files
```

So then I'll invert that grep to eliminate those from the run like this:

```
❯ yarn test --grep-invert '(pool-loads|pool-groups|title-bar-buttons).spec' --list
Listing tests:
  copy-paste.spec.ts:5:3 › Copy Paste Data › paste cleans up after itself
  copy-paste.spec.ts:15:3 › Copy Paste Data › pasting data loads into pool
  copy-paste.spec.ts:24:3 › Copy Paste Data › pasting multiple times
  export.spec.ts:37:9 › Export tests › Exporting in Arrow IPC Stream format succeeds
  export.spec.ts:37:9 › Export tests › Exporting in CSV format succeeds
  export.spec.ts:37:9 › Export tests › Exporting in JSON format succeeds
  export.spec.ts:37:9 › Export tests › Exporting in NDJSON format succeeds
  export.spec.ts:37:9 › Export tests › Exporting in TSV format succeeds
  export.spec.ts:37:9 › Export tests › Exporting in VNG format succeeds
  export.spec.ts:37:9 › Export tests › Exporting in Zeek format succeeds
  export.spec.ts:37:9 › Export tests › Exporting in ZJSON format succeeds
  export.spec.ts:37:9 › Export tests › Exporting in ZNG format succeeds
  export.spec.ts:37:9 › Export tests › Exporting in ZSON format succeeds
  histogram.spec.ts:16:7 › Histogram Spec › Histogram appears for zeek data
  ingest.spec.ts:44:9 › Ingest tests › query000: "* | count()"
  ingest.spec.ts:44:9 › Ingest tests › query001: "* | count() by _path | sort _path"
  ingest.spec.ts:44:9 › Ingest tests › query002: "_path=="conn" | count()"
  ingest.spec.ts:44:9 › Ingest tests › query003: "_path=="conn" | cut ts, id.orig_h, id.orig_p, id.resp_h, id.resp_p, proto | sort ts"
  ingest.spec.ts:44:9 › Ingest tests › query004: "_path=="x509" or _path=="ssl" | sort _path"
  no-pool-specified.spec.ts:16:7 › No Pool Specified State › Query missing a from with no pools
  no-pool-specified.spec.ts:22:7 › No Pool Specified State › Query missing a from with some pools
  preview-and-load.spec.ts:5:3 › Preview & Load › create new pool, change key, type <enter>
  queries.spec.ts:18:7 › Query tests › session queries are the default and ordered properly in history
  queries.spec.ts:36:7 › Query tests › named queries' creation, modification, update/save, proper outdated status display
  queries.spec.ts:72:7 › Query tests › named query, save as => new named query
  table.spec.ts:16:7 › Table Testing › named type shows columns
  zed-events.spec.ts:14:7 › Handle Zed server events › pool-new/update/delete/commit
Total: 27 tests in 9 files
```

Then in the workflow files we will call `yarn e2e:ci`instead of `yarn e2e`.

**Review by jameskerr — APPROVED** [review]

**Comment by philrz** [comment]
Thanks @jameskerr!
